metricflow/dataset/convert_data_source.py

```diff
@@ -353,26 +352,6 @@ def _create_identifier_instances(
                 )
         return identifier_instances, select_columns
 
-    @staticmethod
-    def _find_primary_time_dimension(data_source: DataSource) -> TimeDimensionSpec:
```

Comment:
maybe it's outside of the changes in this PR, but we would still need to check that a data source has a primary time dimension if not all measures have agg_time_dimension set right?

Reply:
Good question! I'm unsure. Specifically I'm unsure if we want to do that in assertions during `convert_data_source` stuff. The validations will catch this, so if they put changes to their model through validations, it won't be a problem.

What we currently do with the a "primary time dimension" is set it to the `agg_time_dimension` for measures that don't have an `agg_time_dimension` set in this [transformation](https://github.com/transform-data/metricflow/blob/33f18ac5ad47b1df08c694316849c82e3ea2cf50/metricflow/model/transformations/agg_time_dimension.py#L24). Then if the model goes through validations, it'll get caught by the above rule. If it doesn't go through validations it will fail somewhere. Specifically if something tries to get the `agg_time_dimension` directly and do something with it, a error involving something about `None` will be raised. However we do have [checked_agg_time_dimension](https://github.com/transform-data/metricflow/blob/33f18ac5ad47b1df08c694316849c82e3ea2cf50/metricflow/model/objects/elements/measure.py#L39) which will raise an assertion error (which is what the the `convert_data_source` used to do).

Reply:
Basically if the expectation is that an error will get raised at some point for measure that doesn't have an `agg_time_dimension`, I'm confident that is the case with these changes.